kotlinx-coroutines-core/common/src/channels/ArrayChannel.kt

```diff
@@ -23,13 +23,15 @@ internal open class ArrayChannel<E>(
     /**
      * Buffer capacity.
      */
-    val capacity: Int
+    private val capacity: Int,
+    private val onBufferOverflow: BufferOverflow
 ) : AbstractChannel<E>() {
     init {
         require(capacity >= 1) { "ArrayChannel capacity must be at least 1, but $capacity was specified" }
```

Comment:
By the way, since this class is internal, we can replace `require` with `assert` or even remove this check completely

Reply:
👍 Replaced with `assert { ... }`

Reply:
Rolled back, added comment. It is actually used by `Channel` constructor function. Instead of checking for invalid capacity there is seems safer and more logic to check it here.